src/3d/qgsrubberband3d.h

```diff
@@ -134,9 +163,12 @@ class _3D_EXPORT QgsRubberBand3D
   private:
     void updateGeometry();
     void updateMarkerMaterial();
+    void setupMarker( Qt3DCore::QEntity *parentEntity );
+    void setupLine( Qt3DCore::QEntity *parentEntity, QgsWindow3DEngine *engine );
+    void setupPolygon( Qt3DCore::QEntity *parentEntity );
 
-  private:
     QgsLineString mLineString;
+    QgsPolygon mPolygon;
```

Comment:
worth consideration - rather than having both QgsLineString and QgsPolygon, why not instead have QgsGeometry as a container (which could be polygon, linestring, potentially in the future also multi-polygon or multi-linestring) - then at least we do not need to keep two representations of the same point data - and just in updateGeometry() we would extract the polygon / linestring data as needed?